In [1]:
import os
import shutil

In [7]:
#Path
source_root = r"C:\Users\KARISHMA\Desktop\deep learning\Depth_Estimation\dataset\data\nyu2_train\basement_0001a_out" #"data/nyu2_train/basement_001a_out"

In [9]:
print(os.path.exists(source_root))
print(len(os.listdir(source_root)))
print(os.listdir(source_root)[:5])

True
562
['1.jpg', '1.png', '10.jpg', '10.png', '100.jpg']


In [12]:
scene_path = os.path.join(source_root, os.listdir(source_root)[0])
print(scene_path)
print(os.listdir(scene_path)[:10])

C:\Users\KARISHMA\Desktop\deep learning\Depth_Estimation\dataset\data\nyu2_train\basement_0001a_out\1.jpg


NotADirectoryError: [WinError 267] The directory name is invalid: 'C:\\Users\\KARISHMA\\Desktop\\deep learning\\Depth_Estimation\\dataset\\data\\nyu2_train\\basement_0001a_out\\1.jpg'

In [10]:
#output folder
output_root = "data"
images_out = os.path.join(output_root, "images")
depth_out = os.path.join(output_root, "depth")

os.makedirs(images_out, exist_ok=True)
os.makedirs(depth_out, exist_ok=True)

In [14]:
counter = 0

for scene in os.listdir(source_root):
    scene_path = os.path.join(source_root, scene)

    if not os.path.isdir(scene_path):
        continue
    files = os.listdir(scene_path)

    jpg_files = [f for f in files if f.endswith(".jpg")]

    for file in jpg_files:
        base_name = os.path.splitext(jpg)[0]
        png_name = base_name + ".png"

        if png_name in files:
            rgb_path = os.path.join(scene_path, jpg)
            depth_path = os.path.join(scene_path, png_name)

            new_name = f"{counter:06d}"

            shutil.copy(rgb_path, os.path.join(images_out, new_name + ".jpg"))
            shutil.copy(depth_path, os.path.join(depth_out, new_name + ".png"))

            counter += 1

        """
        if file.endswith(".jpg"):
            rgb_path = os.path.join(scene_path, file)
            depth_path = os.path.join(scene_path, file.replace(".jpg", ".png"))

            if not os.path.exists(depth_path):
                continue

            new_name = f"{counter:06d}"

            shutil.copy(rgb_path, os.path.join(images_out, new_name + ".jpg"))
            shutil.copy(depth_path, os.path.join(depth_out, new_name + ".png"))

            counter += 1
        """
print(f"Done! Total samples: {counter}")


Done! Total samples: 0


In [16]:
import os
import shutil

source_root = r"C:\Users\KARISHMA\Desktop\deep learning\Depth_Estimation\dataset\data\nyu2_train"

output_root = "data"
images_out = os.path.join(output_root, "images")
depth_out = os.path.join(output_root, "depth")

os.makedirs(images_out, exist_ok=True)
os.makedirs(depth_out, exist_ok=True)

counter = 0

for scene in os.listdir(source_root):
    scene_path = os.path.join(source_root, scene)

    # Skip if not a folder
    if not os.path.isdir(scene_path):
        continue

    files = os.listdir(scene_path)

    jpg_files = [f for f in files if f.endswith(".jpg")]

    for jpg in jpg_files:
        base_name = os.path.splitext(jpg)[0]
        png_name = base_name + ".png"

        if png_name in files:
            rgb_path = os.path.join(scene_path, jpg)
            depth_path = os.path.join(scene_path, png_name)

            new_name = f"{counter:06d}"

            shutil.copy(rgb_path, os.path.join(images_out, new_name + ".jpg"))
            shutil.copy(depth_path, os.path.join(depth_out, new_name + ".png"))

            counter += 1

print(f"Done! Total samples: {counter}")

Done! Total samples: 50688


In [17]:
import os
import shutil

source_root = r"C:\Users\KARISHMA\Desktop\deep learning\Depth_Estimation\dataset\data\nyu2_test"

output_root = "data"
images_out = os.path.join(output_root, "images")
depth_out = os.path.join(output_root, "depth")

os.makedirs(images_out, exist_ok=True)
os.makedirs(depth_out, exist_ok=True)

counter = 0

for scene in os.listdir(source_root):
    scene_path = os.path.join(source_root, scene)

    # Skip if not a folder
    if not os.path.isdir(scene_path):
        continue

    files = os.listdir(scene_path)

    jpg_files = [f for f in files if f.endswith(".jpg")]

    for jpg in jpg_files:
        base_name = os.path.splitext(jpg)[0]
        png_name = base_name + ".png"

        if png_name in files:
            rgb_path = os.path.join(scene_path, jpg)
            depth_path = os.path.join(scene_path, png_name)

            new_name = f"{counter:06d}"

            shutil.copy(rgb_path, os.path.join(images_out, new_name + ".jpg"))
            shutil.copy(depth_path, os.path.join(depth_out, new_name + ".png"))

            counter += 1

print(f"Done! Total samples: {counter}")

Done! Total samples: 0


In [18]:
import os
import shutil

source_root = r"C:\Users\KARISHMA\Desktop\deep learning\Depth_Estimation\dataset\data\nyu2_test"

output_root = "nyu_test_processed"
images_out = os.path.join(output_root, "images")
depth_out = os.path.join(output_root, "depth")

os.makedirs(images_out, exist_ok=True)
os.makedirs(depth_out, exist_ok=True)

files = os.listdir(source_root)

# Get all RGB files
color_files = [f for f in files if f.endswith("_colors.png")]

counter = 0

for color_file in color_files:
    base_name = color_file.replace("_colors.png", "")
    depth_file = base_name + "_depth.png"

    if depth_file in files:
        rgb_path = os.path.join(source_root, color_file)
        depth_path = os.path.join(source_root, depth_file)

        new_name = f"{counter:06d}"

        shutil.copy(rgb_path, os.path.join(images_out, new_name + ".png"))
        shutil.copy(depth_path, os.path.join(depth_out, new_name + ".png"))

        counter += 1

print(f"Done! Total samples: {counter}")

Done! Total samples: 654


In [19]:
import os

source_root = r"C:\Users\KARISHMA\Desktop\deep learning\Depth_Estimation\dataset\data\nyu2_test"

files = os.listdir(source_root)

color_files = [f for f in files if f.endswith("_colors.png")]
depth_files = [f for f in files if f.endswith("_depth.png")]

print("RGB images:", len(color_files))
print("Depth images:", len(depth_files))

RGB images: 654
Depth images: 654


In [20]:
pairs = 0

for f in color_files:
    base = f.replace("_colors.png", "")
    if base + "_depth.png" in files:
        pairs += 1

print("Matching pairs:", pairs)

Matching pairs: 654


In [21]:
import os

source_root = r"C:\Users\KARISHMA\Desktop\deep learning\Depth_Estimation\dataset\data\nyu2_test"

files = os.listdir(source_root)

# Check specific example
print("01448_colors.png exists:", "01448_colors.png" in files)
print("01448_depth.png exists:", "01448_depth.png" in files)

01448_colors.png exists: True
01448_depth.png exists: True


In [22]:
missing = []

for i in range(1449):
    name = f"{i:05d}"
    if not (name + "_colors.png" in files and name + "_depth.png" in files):
        missing.append(name)

print("Missing pairs:", len(missing))
print("Some missing:", missing[:10])

Missing pairs: 795
Some missing: ['00002', '00003', '00004', '00005', '00006', '00007', '00009', '00010', '00011', '00012']


🔥 KEY INSIGHT (VERY IMPORTANT FOR YOUR REPORT)

👉 Even though indices go till 1448:

✔️ Only ~654 are valid RGB–Depth pairs
✔️ These are the benchmark evaluation set


Our dataset is:
clean
standardmatches research usage

why some images missing: 
might contain frames from video sequences
not all frames have valid depth, proper alignment, clean data....and so is filtered

In [ ]:
#Actual working code format

In [2]:
#import
import os
import shutil

Preprocessing training data

In [3]:
#source root
#source_root = r"C:\Users\KARISHMA\Desktop\deep learning\Depth_Estimation\dataset\data\nyu2_train"
from pathlib import Path

source_root = Path("dataset/data/nyu2_train")

In [ ]:
#output root
output_root = "dataset/preprocessed_data/nyu_train_preprocessed"
images_out = os.path.join(output_root, "images")
depth_out = os.path.join(output_root, "depth")

#making output files
os.makedirs(images_out, exist_ok=True)
os.makedirs(depth_out, exist_ok=True)

In [5]:
#Segregation code

counter = 0

for scene in os.listdir(source_root):
    scene_path = os.path.join(source_root, scene)

    # Skip if not a folder
    if not os.path.isdir(scene_path):
        continue

    files = os.listdir(scene_path)

    jpg_files = [f for f in files if f.endswith(".jpg")]

    for jpg in jpg_files:
        base_name = os.path.splitext(jpg)[0]
        png_name = base_name + ".png"

        if png_name in files:
            rgb_path = os.path.join(scene_path, jpg)
            depth_path = os.path.join(scene_path, png_name)

            new_name = f"{counter:06d}"

            shutil.copy(rgb_path, os.path.join(images_out, new_name + ".jpg"))
            shutil.copy(depth_path, os.path.join(depth_out, new_name + ".png"))

            counter += 1

print(f"Done! Total samples: {counter}")

Done! Total samples: 50688


Preprocessing testing data

In [6]:
#source root
#source_root = r"C:\Users\KARISHMA\Desktop\deep learning\Depth_Estimation\dataset\data\nyu2_test"

source_root = Path("dataset/data/nyu2_test")

In [7]:
#output root
output_root = "dataset/preprocessed_data/nyu_test_processed"
images_out = os.path.join(output_root, "images")
depth_out = os.path.join(output_root, "depth")

os.makedirs(images_out, exist_ok=True)
os.makedirs(depth_out, exist_ok=True)

In [8]:
#segregation
files = os.listdir(source_root)

# Get all RGB files
color_files = [f for f in files if f.endswith("_colors.png")]

counter = 0

for color_file in color_files:
    base_name = color_file.replace("_colors.png", "")
    depth_file = base_name + "_depth.png"

    if depth_file in files:
        rgb_path = os.path.join(source_root, color_file)
        depth_path = os.path.join(source_root, depth_file)

        new_name = f"{counter:06d}"

        shutil.copy(rgb_path, os.path.join(images_out, new_name + ".png"))
        shutil.copy(depth_path, os.path.join(depth_out, new_name + ".png"))

        counter += 1

print(f"Done! Total samples: {counter}")

Done! Total samples: 654


In the nyu2_test file there are images with numbers not in sequence. this could be because they might have taken the images as frames from video sequences. Not all frames might have had valid depth and proper alignment. This shows that our data is clean, is filtered and matches research standards.